model-optimizer/extensions/middle/InterpolateSequenceToInterpolate.py

```diff
@@ -127,37 +230,54 @@ def replace_sequence(seq: List[Node], graph: Graph):
-    axes_of_node = int64_array([z[0] for z in axis_to_size])
-    sizes = int64_array([z[1] for z in axis_to_size])
+    #      (axis, output size for this axis) (opset1)
+    # or
+    #      (axis, output size for this axis, output scales for this axis) (opset4)
+    if seq[0].get_opset() == 'opset1':
+        for interp in seq:
+            dims_and_scales_.extend(zip(Interpolate.get_axes(interp),
+                                        interp.in_port(1).get_connection().get_source().node.value))
+
+        axis_to_size = sorted(list(dict(dims_and_scales_).items()), key=lambda x: x[0])
+        axes_of_node = int64_array([z[0] for z in axis_to_size])
+        sizes = int64_array([z[1] for z in axis_to_size])
+        scales = np.ones(len(axis_to_size))
+    else:
+        for interp in seq:
+            dims_and_scales_.extend(zip(Interpolate.get_axes(interp),
+                                        interp.in_port(1).get_connection().get_source().node.value,
+                                        interp.in_port(2).get_connection().get_source().node.value))
+
+        axis_to_size = sorted(dims_and_scales_, key=lambda x: x[0])
+        axes_of_node = int64_array([z[0] for z in axis_to_size])
+        sizes = int64_array([z[1] for z in axis_to_size])
+        scales = np.array([z[2] for z in axis_to_size])
 
     fst_interp_node = seq[0]
     last_interp_node = seq[-1]
     fst_interp_node_name = fst_interp_node.name
-    fst_interp_node_mode = fst_interp_node.mode
-    fst_interp_node_align_corners = fst_interp_node.soft_get('align_corners', default=0)
-    fst_interp_node_antialias = fst_interp_node.soft_get('antialias', default=0)
-    fst_interp_node_pads_begin = fst_interp_node.soft_get('pads_begin', default=0)
-    fst_interp_node_pads_end = fst_interp_node.soft_get('pads_end', default=0)
-    interp_node = Interpolate(graph, dict(name=fst_interp_node_name + '/Interpolate_',
-                                          axes=axes_of_node,
-                                          mode=fst_interp_node_mode,
-                                          align_corners=fst_interp_node_align_corners,
-                                          antialias=fst_interp_node_antialias,
-                                          pads_begin=fst_interp_node_pads_begin,
-                                          pads_end=fst_interp_node_pads_end)).create_node()
-
-    scales_node = Const(graph, dict(name=fst_interp_node_name + '/scales_', value=sizes)).create_node()
-    scales_node.out_port(0).connect(interp_node.in_port(1))
-
-    fst_interp_connection = fst_interp_node.in_port(0).get_connection()
-    fst_interp_connection.set_destination(interp_node.in_port(0))
-
-    last_interp_node.out_port(0).get_connection().set_source(interp_node.out_port(0))
+    attributes = get_interpolate_attributes(fst_interp_node)
+    attributes['name'] = fst_interp_node_name + '/Interpolate_'
```

Comment:
This transformation does not keep the original node name.

Reply:
Fixed.